client/src/lib.rs

```diff
@@ -80,9 +80,9 @@ pub struct ClientOptions {
     /// in all RPC calls. The server decides if the client is supported based on this.
     pub client_version: String,
 
-    /// Other non-required static headers which should be attached to every request
+    /// Other non-required headers which should be attached to every request
     #[builder(default)]
-    pub static_headers: HashMap<String, String>,
+    pub headers: Arc<Mutex<HashMap<String, String>>>,
```

Comment:
Use a parking_lot mutex to avoid fallible locking

Reply:
I also don't really love the idea of this being what gets mutated. IMO the options should remain static.

We could change this to be `default_headers` and then put this as a field on the interceptor specifically named `header_overrides` and that can be changed with a function call.

Reply:
Can you explain why that's better?